modules/core/efs/stack_efs.py

```diff
@@ -0,0 +1,79 @@
+import os
+from typing import Any, cast
+
+import cdk_nag
+from aws_cdk import Aspects, RemovalPolicy, Stack, Tags
+from aws_cdk import aws_ec2 as ec2
+from aws_cdk import aws_efs as efs
+from aws_cdk import aws_iam as iam
+from constructs import Construct, IConstruct
+
+project_dir = os.path.dirname(os.path.abspath(__file__))
+
+
+class EFSFileStorage(Stack):
+    def __init__(
+        self,
+        scope: Construct,
+        id: str,
+        deployment_name: str,
+        module_name: str,
+        vpc_id: str,
+        **kwargs: Any,
+    ) -> None:
+
+        super().__init__(
+            scope,
+            id,
+            description="This stack creates an EFS filesystem",
+            **kwargs,
+        )
+
+        self.deployment_name = deployment_name
+        self.module_name = module_name
+        Tags.of(scope=cast(IConstruct, self)).add(key="Deployment", value=f"addf-{self.deployment_name}")
+
+        dep_mod = f"addf-{self.deployment_name}-{self.module_name}"
+        dep_mod = dep_mod[:30]
+
+        self.vpc_id = vpc_id
+        self.vpc = ec2.Vpc.from_lookup(
+            self,
+            "VPC",
+            vpc_id=self.vpc_id,
+        )
+
+        self.efs_security_group = ec2.SecurityGroup(self, "EFSSecurityGroup", vpc=self.vpc, allow_all_outbound=True)
+        self.efs_filesystem = efs.FileSystem(
+            self,
+            "Filesystem",
+            vpc=self.vpc,
+            security_group=self.efs_security_group,
+            removal_policy=RemovalPolicy.DESTROY,
```

Comment:
This we need to pass through as parameter and have the default to retain. Destroy is very dangerous especially if someone isn't aware that it is the default.

Reply:
Completed